collectors/mesos/agent/agent.go

```diff
@@ -123,6 +123,19 @@ func (c *Collector) metricsMessages() (out []producers.MetricsMessage) {
 			continue
 		}
 
+		ei, ok := getExecutorInfoByExecutorID(cm.ExecutorID, fi.Executors)
+		if !ok {
+			c.log.Warnf("Did not find ExecutorInfo for executor ID %s, skipping!", cm.ExecutorID)
+			continue
+		}
+
+		ti, ok := getTaskInfoByContainerID(cm.ContainerID, ei.Tasks)
+		if !ok {
+			// This is not a warning because task ID is not guaranteed to be set, eg
+			// a custom executor is a container with no associated task.
+			c.log.Debugf("Did not find TaskInfo for container ID %s, skipping!", cm.ContainerID)
```

Comment:
In that case, should we also not skip this `cm` if the executor info could not be found?